pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -124,27 +147,50 @@ func (s *FlinkStateMachine) handle(ctx context.Context, application *v1alpha1.Fl
 	if !v1alpha1.IsRunningPhase(application.Status.Phase) {
 		logger.Infof(ctx, "Handling state %s for application", application.Status.Phase)
 	}
-
-	switch application.Status.Phase {
-	case v1alpha1.FlinkApplicationNew, v1alpha1.FlinkApplicationUpdating:
-		// Currently just transitions to the next state
-		return s.handleNewOrUpdating(ctx, application)
-	case v1alpha1.FlinkApplicationClusterStarting:
-		return s.handleClusterStarting(ctx, application)
-	case v1alpha1.FlinkApplicationSubmittingJob:
-		return s.handleSubmittingJob(ctx, application)
-	case v1alpha1.FlinkApplicationRunning, v1alpha1.FlinkApplicationDeployFailed:
-		return s.handleApplicationRunning(ctx, application)
-	case v1alpha1.FlinkApplicationSavepointing:
-		return s.handleApplicationSavepointing(ctx, application)
-	case v1alpha1.FlinkApplicationRollingBackJob:
-		return s.handleRollingBack(ctx, application)
-	case v1alpha1.FlinkApplicationDeleting:
-		return s.handleApplicationDeleting(ctx, application)
+	if s.IsTimeToHandlePhase(application) {
+		var appErr error
+		switch application.Status.Phase {
+		case v1alpha1.FlinkApplicationNew, v1alpha1.FlinkApplicationUpdating:
+			// Currently just transitions to the next state
+			appErr = s.handleNewOrUpdating(ctx, application)
+		case v1alpha1.FlinkApplicationClusterStarting:
+			appErr = s.handleClusterStarting(ctx, application)
+		case v1alpha1.FlinkApplicationSubmittingJob:
+			appErr = s.handleSubmittingJob(ctx, application)
+		case v1alpha1.FlinkApplicationRunning, v1alpha1.FlinkApplicationDeployFailed:
+			appErr = s.handleApplicationRunning(ctx, application)
```

Comment:
I think we need different logic for the running phases. RIght now if we hit some issues during running (e.g., talking to the JM or k8s api server) we'll quickly exhaust all of our retries, and the operator will refuse to do anything until the application is deleted and recreated. If we're in running, we probably don't want to invoke any of the error handling stuff I think.

Reply:
That makes sense to me. I've added a check to not invoke the error handling code when the application is in a running phase.